dts/bindings/i2c/cdns,i2c-r1p14.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2025 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Cadence I2C controller version 1.4.
+
+  Example I2C device tree node:
+    i2c@e0004000 {
+        compatible = "cdns,i2c-r1p14";
+        clocks = <&clkc 38>;
+        interrupts = <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
+        reg = <0xe0004000 0x1000>;
+        clock-frequency = <400000>;
+        #address-cells = <1>;
+        #size-cells = <0>;
+        fifo-depth = <8>;
+    };
+
+compatible: "cdns,i2c-r1p14"
+
+include: [i2c-controller.yaml, reset-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    required: true
+    description: Reference Input clock for I2C core.
+
+  interrupts:
+    required: true
+
+  clock-frequency:
+    description: Desired I2C bus clock frequency in Hz.
+    enum: [100000, 400000]
```

Comment:
I don't see why this needs to be fixed to only be 100kHz or 400kHz.

Reply:
That's what has been tested. In Linux DT we are just saying max 400kHz. If someone wants to run it on different speed they can come and extend binding with that. But I can't remember anybody who has tested different frequencies other then 100 and 400kHz.

Reply:
lets keep this discussion in one thread